File: react-native-pytorch-core/android/src/main/java/org/pytorch/rn/core/canvas/ImageData.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package org.pytorch.rn.core.canvas;

import android.graphics.Bitmap;

public class ImageData {

  private final int mScaledWidth;
  private final int mScaledHeight;
  private final Bitmap mBitmap;

  public ImageData(Bitmap bitmap, int scaledWidth, int scaledHeight) {
    mBitmap = bitmap;
    mScaledWidth = scaledWidth;
    mScaledHeight = scaledHeight;
  }

  public int getWidth() {
    return mBitmap.getWidth();
  }

  public int getHeight() {
    return mBitmap.getHeight();
  }

  public Bitmap getBitmap() {
    return mBitmap;
  }

  public Bitmap getScaledBitmap() {
    return Bitmap.createScaledBitmap(mBitmap, mScaledWidth, mScaledHeight, false);
  }
}

File: react-native-pytorch-core/ios/Canvas/ImageData.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation

class ImageData {
    enum ImageDataError: Error {
        case scaleImage
    }

    let bitmap: CGImage
    let scaledWidth: CGFloat
    let scaledHeight: CGFloat

    init(bitmap: CGImage, scaledWidth: CGFloat, scaledHeight: CGFloat) {
        self.bitmap = bitmap
        self.scaledWidth = scaledWidth
        self.scaledHeight = scaledHeight
    }

    func getScaledBitmap() throws -> CGImage {
        let size = CGSize(width: self.scaledWidth, height: self.scaledHeight)
        guard let scaledImage = UIImage(cgImage: self.bitmap).resizeImage(size: size),
              let scaledBitmap = scaledImage.cgImage else {
            throw ImageDataError.scaleImage
        }
        return scaledBitmap
    }
}
